examples/booking_app/lib/feature_hotel/search/blocs/hotel_list_bloc.dart

```diff
@@ -105,29 +84,27 @@ class HotelListBloc extends $HotelListBloc {
     ),
   );
 
-  final _searchFilterSubject =
-      BehaviorSubject<HotelSearchFilters>.seeded(HotelSearchFilters());
-  final _querySubject = BehaviorSubject<String>.seeded('');
-  final _dateRangeSubject = BehaviorSubject<DateTimeRange?>.seeded(null);
-
   @override
   Stream<PaginatedList<Hotel>> get searchedHotels => _hotels;
 
   @override
   void dispose() {
-    _searchFilterSubject.close();
-    _querySubject.close();
-    _dateRangeSubject.close();
     _hotels.close();
     super.dispose();
   }
 
   @override
   Stream<String> get query =>
-      _searchFilters.map((filters) => filters.query).shareReplay(maxSize: 1);
+      filters.map((filters) => filters.query).shareReplay(maxSize: 1);
 
   @override
-  Stream<HotelSearchFilters> get filters => _searchFilters;
+  Stream<HotelSearchFilters> get filters => Rx.combineLatest2(
+      _$filterByQueryEvent.delayInput(),
+      _$filterByDateRangeEvent.startWith(null).distinct(),
```

Comment:
@DDavidPrime what about making `_$filterByDateRangeEvent` behaviour subject ? In this way we will keep the filter events consistent

Reply:
After some testing it seems that changing the event to:
```
@RxBlocEvent(type: RxBlocEventType.behaviour, seed: null)
  void filterByDateRange(DateTimeRange? dateRange);
```
will not work as there is an issue with the RxBlocGenerator saying in the generated file the following:
```
The argument type 'DateTimeRange?' can't be assigned to the parameter type 'Null'.
```
After this is fixed, we may change it to the first one as proposed.

Reply:
Please file an issue for this, so we can fix it later. As for now I've just added the type manually ;)